React/Modules/RCTTiming.m

```diff
@@ -181,22 +192,31 @@ - (void)didUpdateFrame:(__unused RCTFrameUpdate *)update
   // Call timers that need to be called
   if (timersToCall.count > 0) {
     [_bridge enqueueJSCall:@"JSTimersExecution.callTimers" args:@[timersToCall]];
-
-    // If we call at least one timer this frame, don't switch to a paused state yet, so if
-    // in response to this timer another timer is scheduled, we don't pause and unpause
-    // the displaylink frivolously.
-    return;
   }
 
-  // No need to call the pauseCallback as RCTDisplayLink will ask us about our paused
-  // status immediately after completing this call
-  if (_timers.count == 0) {
-    _paused = YES;
+  if (_sendIdleEvents) {
+    NSTimeInterval frameElapsed = (CACurrentMediaTime() - update.timestamp) * 1000;
```

Comment:
Not sure if there is a better way to convert this value to a unix timestamp, sometimes it is not in sync with the JS `Date.now` as I get some frame time left greater that 17ms (varies from 15-17.5).

`CACurrentMediaTime() - update.timestamp` is always positive so this is not the issue. I suspect that `[[NSDate date] timeIntervalSince1970]` gets out of sync with JS `Date.now`.